server/src/main/java/org/elasticsearch/index/mapper/ParsedDocument.java

```diff
@@ -83,6 +83,13 @@ public void updateSeqID(long sequenceNumber, long primaryTerm) {
         this.seqID.primaryTerm.setLongValue(primaryTerm);
     }
 
+    ParsedDocument toTombstone() {
```

Comment:
maybe call this "markAsSoftDeleted"? NoOps doc are not really tombstones.

Reply:
Yes, noop docs are not really tombstones but `markAsSoftDeleted` is not correct. Is it ok for us to call noop docs  "noop tombstones"?

Reply:
same confusion. Please ignore. I'll come up with better name than tombstone, if I can, but don't let that stop you.